blindsign/blindrsa/blindrsa.go

```diff
@@ -0,0 +1,213 @@
+package blindrsa
+
+// This package implements the blind RSA protocol based on the CFRG specification:
+// https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-rsa-blind-signatures-02
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/subtle"
+	"errors"
+	"hash"
+	"io"
+	"math/big"
+
+	"github.com/cloudflare/circl/blindsign"
+)
+
+// An RSAVerifier represents a Verifier in the RSA blind signature protocol.
+// It carries state needed to produce and validate an RSA blind signature.
+type RSAVerifier struct {
+	// Public key of the Signer
+	pk *rsa.PublicKey
+
+	// Hash function used in producing the message signature
+	hash hash.Hash
+}
+
+// NewRSAVerifier creates a new RSAVerifier using the corresponding Signer parameters.
+func NewRSAVerifier(pk *rsa.PublicKey, hash hash.Hash) RSAVerifier {
+	return RSAVerifier{
+		pk:   pk,
+		hash: hash,
+	}
+}
+
+func encodeMessageEMSAPSS(message []byte, key *rsa.PublicKey, hash hash.Hash, salt []byte) ([]byte, error) {
+	hash.Write(message)
+	digest := hash.Sum(nil)
+	hash.Reset()
```

Comment:
nit: This assumes that the hash state was initialized before calling this method. While this assumption is valid with this code currently, in my experience with the hash.Hash API, this is a footgun. I would move this to the top of this function.